### PR TITLE
WIP: Arm64 build support

### PR DESCRIPTION
**Pull request by scottmalkie**
### Description
With the GA of EKS support for Graviton2 last month, it would be nice to have the `eksctl` binary and docker image built for Arm64. As `eksctl` is 97%+ written in Go, this should be mostly a build/script/packaging change rather than substantive code changes.

- Added another option to `install-build-deps.sh` that determines if compiling on arm64 OS.
- Added arm64 options to `goreleaser` yamls

Issue #2621 (current)
Issues #2280 , #1624 (closed)

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [X] Manually tested
- [X] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [X] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: build/scripts/install-build-deps.sh [hunk 1]
```diff
@@ -9,6 +9,8 @@ fi
 
 if [ "$(uname)" = "Darwin" ] ; then
   OSARCH="darwin-amd64"
+elif [ "$(uname -i)" = "aarch64" ] ; then
```
Comment (sayboras) [inline build/scripts/install-build-deps.sh:12]: Understand this is existing logic, however, I am curious why these two parameters are requried for `go install`. Will go env GOARCH and GOOS be assgined automatically ?

Comment (scottmalkie) [reply build/scripts/install-build-deps.sh:12]: GOOS and GOARCH _should_ be assigned based on build host architecture but it seems the "OSARCH" env var is used elsewhere, as I can build successfully on arm64 (Graviton2) with it, but not without it.

Comment (scottmalkie) [reply build/scripts/install-build-deps.sh:12]: I don't have the exact error in my notes, let me try again and document it.

Comment (sayboras) [reply build/scripts/install-build-deps.sh:12]: yup, I tried to search OSARCH in repo, but can't find any place using this variable.

**Comment by sayboras** [comment]
@scottmalkie my apology, I didn't see this PR before creating https://github.com/weaveworks/eksctl/pull/2725. Happy to close mine and keep yours.

**Comment by scottmalkie** [comment]
> @scottmalkie my apology, I didn't see this PR before creating #2725. Happy to close mine and keep yours.

Oh no, yours is much better. I'll close this one.  Whichever one gets an arm64 binary out there first I will wholeheartedly support. :)
